### ISSUE TITLE
Chromebook: Prompted for Password on SSH

### ISSUE THREAD

**Issue by Amarjit3** [termux/termux-app#967]
I am attempting to log in to Termux on my Chromebook using NaSSH in accordance with the instructions [here](https://blog.lessonslearned.org/building-a-more-secure-development-chromebook/). 

I installed the app on my Asus C302 chromebook. I then generated the public and private SSH keys and amended the authorized_keys file accordingly. I am now attempting to log in to Termux using the NaSSH app from the Chrome store. I followed the termux-chroot and SSHD commands as per the instructions.

I have determined what my username is (using whoami), IP address and port number. When I enter these into NaSSH and use the id_rsa.pub key it successfully connects but I am asked for a password. I have tried setting a password using passwd in Termux itself to no avail.

When I do not use the keys and use the Termux password I am also unsuccessful. I also tried removing the requirement for a password in the server config files with no success.

Is anyone able to help? It is possible to use NaSSH judging from [this](https://medium.com/@clumsycontraria/learning-to-code-on-a-bone-stock-chromebook-a7d0e75303bb) blog also.

**Comment by Grimler91** [comment]
I'm not familiar with NaSSH (and I don't own a chromebook) so I can only give generic tips:

> amended the authorized_keys file

Just to be sure, you amended to termux's authorized_keys, right?

> I followed the termux-chroot and SSHD commands

I have no idea why that blog tells you to use termux-chroot. It should be possible to run sshd without termux-chroot (so please ignore that part of the instructions).

> I have determined what my username is (using whoami), IP address and port number

There's no need to specify username, it is ignored since termux is a single user environment, also the port is 8022 unless you've made changes to the default config files. Also how did you determine the ip? The failing password prompt sort of sounds like you are trying to ssh from the chromebook to the chromebook (with the termux user/pw).

> When I enter these into NaSSH and use the id_rsa.pub key

That seems weird, you should probably specify the id_rsa file. The id_rsa.pub file is the one you need to append to termux's authorized_hosts.

Next step for understanding the problem should be to look at the logs from sshd and ssh.
You can run `sshd -E ~/sshd.log` to get a logfile (be sure to kill the old sshd daemon before starting the new one). Maybe NaSSH has some way to view the logs for as well?

**Comment by xeffyr** [comment]
@Amarjit3 Have you read Termux Wiki about this ?

Before messing with `termux-chroot` and `sshd`, I suggest to try instructions on wiki.

**About SSH on Termux Wiki:**
* https://wiki.termux.com/wiki/Remote_Access#Setting_up_password_authentication - password authentication.
* https://wiki.termux.com/wiki/Remote_Access#Setting_up_public_key_authentication - pubkey authentication.

**Comment by Amarjit3** [comment]
Many thanks for both of your quick responses. It sounds like the blog instructions may not be correct. I will follow the tutorials tomorrow and provide some logs where possible.

@Grimler91 - you are essentially correct. The essence of what I am trying to do (the same as the two blogs) is to run Termux on the Chromebook which runs in Android space and SSH into Termux using the NaSSH app which runs in 'Chrome space' on the same machine. 

Hopefully following these tutorials will show that it is possible.

**Comment by seisdr** [comment]
God forgive you son

**Comment by Amarjit3** [comment]
I am attempting to SSH into Termux using the Chrome SSH app, installed on the same machine (Asus C302 chromebook). To recap, I did a clean install of Termux and carried out the instructions below:
`
apt update   
apt upgrade
pkg install proot openssh
`
Then, as I need access to my Downloads folder on my Chromebook:
`
termux-setup-storage
`
As per the instructions in the blog I created a directory that could store the SSH keys:
`
termux-chroot
mkdir -p /storage/emulated/0/Download/devbox/ssh   
ln -s /storage/emulated/0/Download/devbox /home/devbox
`
I then exited chroot and I generated the SSH keys:
ssh-keygen -t rsa -b 4096 -f  /home/devbox/ssh/id_rsa
`
The public key is then added to authorized_keys file:
`
cat /home/devbox/ssh/id_rsa.pub >> /home/.ssh/authorized_keys
`
To find my IP address and user name I used:
`
whoami
ifconfig arc0 | awk '/inet /{print $2}'
`
I then started sshd.

I am able to login using SSH, however, I am asked for a password which I cannot avoid (as per my first post). I have to specify a user on the app (it won't let be log in otherwise). The FAQ also says to include BOTH keys (which I did). There is no log on the Chrome SSH app which I can access but I looked at my log on Termux and found:

Bind to port on 8022 on :: failed Address already in use
Bind to port on 8022 failed. Address already in use
Cannot bind any address

I had to use chroot to create the symlinks and write the keys to the local storage on the Chromebook but did not need it to start sshd.

As per the responses to my to my first post, I then tried following the tutorials and got the same errors but this time I could not even log in through SSH (the Chrome app said could not connect).

Is anyone able to advise or has accomplished this?

**Comment by Grimler91** [comment]
```
Bind to port on 8022 on :: failed Address already in use
Bind to port on 8022 failed. Address already in use
Cannot bind any address
```
Means that you didn't kill the old daemon before starting the logging one. Run `ps` and `kill <PID>` where <PID> is the PID of sshd as outputed from `ps`. You can then run `sshd -E ~/sshd.log`.

> I had to use chroot to create the symlinks and write the keys to the local storage on the Chromebook but did not need it to start sshd.

chroot doesn't grant you access to anything, it just replaces /data/data/com.termux/files/home /home (and similar) in your commands. 
Your command `ln -s /storage/emulated/0/Download/devbox /home/devbox` in chroot is the same as running `ln -s /storage/emulated/0/Download/devbox /data/data/com.termux/files/home/devbox` outside of chroot.

**Comment by Amarjit3** [comment]
Thank you both so much for your help.

I carried out the following commands as suggested

> pkg install openssh
> passwd

I then made sure there was no sshd running as below:

> Means that you didn't kill the old daemon before starting the logging one. Run ps and kill <PID> where is the PID of sshd as outputed from ps.

I then logged in using the Chrome SSH app using the IP address (localhost doesn't work), username and port. I did have to reinstall the app but it works fine. The Android SSH app I have also works (ConnectBot) using localhost rather than the IP address. 

Your help is much appreciated!!
